recipes/approvaltests.cpp/8.0.0/test_package/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+cmake_minimum_required(VERSION 2.8)
+PROJECT(test_package)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
```

Comment:
We try to avoid these lines in the `CMakeLists.txt`. If something is needed, it should go in the Conan recipe and Conan will pass the proper value to CMake. I know this is [an open issue](https://github.com/conan-io/conan-center-index/issues/54) yet (it is blocking some recipes that require 14 or 17, other use this CMake variables), but in this case, as it is C++11, it is the default for some compilers and it will work. 

Nevertheless, if this is needed, I agree with it until we have a proper solution for this issue.

Reply:
ApprovalTests.cpp does depend on C++11. I think the correct thing to do is to set the target properties in the ApprovalTests.cpp CMakelists.txt, but that will be in a future version.

Reference: https://crascit.com/2015/03/28/enabling-cxx11-in-cmake/

I'm open to other proposals, but for now, it looks like this is required.

Reply:
Can you set it in the tests?

I've recently seen things like (pseudo code)
if (not CMAKE_CXX_STANDARD)
    set(CMAKE_CXX_STANDARD 11)
....
endif()

Which seems good, but I've no idea if it fits in the context of what you are working on.

Reply:
I'm not sure that setting it in the test code is better than in the CMake file @claremacrae . This only affects the package tests. It doesn't propagate to users.

Reply:
OK, glad to hear it's about tests... I meant that I had seen those lines in cmake code... so if people run CMake and define CMAKE_CXX_STANDARD on the command line, their value stands... And if they don't, then the test's CMake file sets the value itself...

But I am on seriously thin ice here, knowing so little about Conan...

Think I'll step away from the keyboard for now!

Reply:
The C++ standard is an open issue in Conan. Probably something we will talk about during [ConanDays](https://conandays.conan.io/). We think that the package manager should know about the standard required by the libraries in order to resolve the graph and ensure it is compatible before starting to download and build all the things... 

Following this reasoning, we should use `compiler.cppstd` in the profiles to define the version we want to use and Conan will pass that value to CMake (or any other build system). But, on the other hand, majors compilers declare ABI compatibility between different stable versions of the standard (although there are some issues for VS), so if all those libraries are ABI compatible, this information shouldn't be encoded in the packageID and Conan can safely build a graph with versions mixed (just [checking a `minimum_cppstd_required`](https://github.com/conan-io/conan-center-index/pull/558/files#diff-81be493f6e19a50c042fd55e58c60d29R23) would be enough for libraries requiring a minimum)

...but then, in Conan Center we should generate profiles for all the `cppstd` values for each compiler, and it would increase _a lot_ the number of binaries to generate.

Now the information is scattered around, probably a good reference to start reading about it (Conan related) is https://github.com/conan-io/conan/issues/6282

Summing up, there are only personal opinions right now, not even a recommendation from our side, so it's up to the community to use the implementation that works and we'll try to learn from it.